### PR TITLE
chore: update min go version to 1.22 and update to the 1.23 toolchain

### PR DESCRIPTION
**Pull request by machine424**
enforce the go version in test_go_oldest through `GOTOOLCHAIN=local`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module github.com/prometheus/prometheus
 
-go 1.21.0
+go 1.22.0
 
-toolchain go1.22.5
+toolchain go1.23.0
```
Comment (machine424) [inline go.mod:5]: Now that I'm thinking about it, with toolchain, `Go tests with previous Go version` are also using the latest version now, I think they need a `GOTOOLCHAIN=local`

https://github.com/prometheus/prometheus/actions/runs/10471451070/job/28998762258?pr=14697#step:4:5

**Comment by machine424** [comment]
last retry
/prombench restart main

File: go.mod (hunk 1 above)
Comment (machine424) [reply go.mod:5]: Done

**Comment by machine424** [comment]
cc @SuperQ

**Review by SuperQ — APPROVED** [review]
Thanks!

**Comment by machine424** [comment]
The dashboards provide no indication of notable regression.

**Pull request merged**